Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp

```diff
@@ -2942,6 +3032,40 @@ B3::Value* B3IRGenerator::createCallPatchpoint(BasicBlock* block, Origin origin,
     return patchpoint;
 }
 
+void B3IRGenerator::createReturnCallPatchpoint(BasicBlock* block, Origin origin, const Vector<ArgumentLocation>& locations, Vector<ExpressionType>& args, const Checked<int32_t> &newStackOffset, const ScopedLambda<void(PatchpointValue*, Box<PatchpointExceptionHandle>)>& patchpointFunctor)
+{
+    Vector<B3::ConstrainedValue> constrainedArguments;
+
+    // NOTE: We reuse the caller stack space and spill the exceeding arguments to avoid corrupting slots below the frame pointer. They will be moved to the correct locations in the generator.
```

Comment:
Slots belonging to the previous frame might be less ambiguous? Why can't we expand sp here first?

Reply:
To not clobber callee saves and slots used by the register allocator? See my previous comment on `createReturnCallPatchpoint`.